mixer/template/tracespan/tracespan.proto

```diff
@@ -0,0 +1,103 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package traceSpan;
+
+import "google/protobuf/timestamp.proto";
+import "mixer/v1/config/descriptor/value_type.proto";
+import "mixer/v1/template/extensions.proto";
+
+option (istio.mixer.v1.template.template_variety) = TEMPLATE_VARIETY_REPORT;
+
+// TraceSpan represents an individual span within a distributed trace.
+//
+// When writing the configuration, the value for the fields associated with this template can either be a
+// literal or an [expression](https://istio.io/docs/reference/config/mixer/expression-language.html). Please note that if the datatype of a field is not istio.mixer.v1.config.descriptor.ValueType,
+// then the expression's [inferred type](https://istio.io/docs/reference/config/mixer/expression-language.html#type-checking) must match the datatype of the field.
+//
+// Example config:
+// ```
+// apiVersion: "config.istio.io/v1alpha2"
+// kind: tracespan
+// metadata:
+//   name: default
+//   namespace: istio-system
+// spec:
+//   traceId: request.headers["x-b3-traceid"]
+//   spanId: request.headers["x-b3-spanid"] | ""
+//   parentSpanId: request.headers["x-b3-parentspanid"] | ""
+//   spanName: request.path | "/"
+//   startTime: request.time
+//   endTime: response.time
+//   spanTags:
+//     http.method: request.method | ""
+//     http.status_code: response.code | 200
+//     http.url: request.path | ""
+//     request.size: request.size | 0
+//     response.size: response.size | 0
+//     source.ip: source.ip | ip("0.0.0.0")
+//     source.service: source.service | ""
+//     source.user: source.user | ""
+//     source.version: source.labels["version"] | ""
+// ```
+//
+// See also: [Distributed Tracing](https://istio.io/docs/tasks/telemetry/distributed-tracing.html)
+// for information on tracing within Istio.
+message Template {
+    // Trace ID is the unique identifier for a trace. All spans from the same
+    // trace share the same Trace ID.
+    //
+    // Required.
+    string trace_id = 1;
+
+    // Span ID is the unique identifier for a span within a trace. It is assigned
```

Comment:
Is this auto assigned if not provided ?

Reply:
Yes. Although, I think that is an adapter detail. Certainly, proper implementations will auto-assign.